src/jit.rs

```diff
@@ -393,7 +399,7 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
         self.emit_subroutines();
 
         while self.pc * ebpf::INSN_SIZE < self.program.len() {
-            if self.offset_in_text_section + MAX_MACHINE_CODE_LENGTH_PER_INSTRUCTION > self.result.text_section.len() {
+            if self.offset_in_text_section + MAX_MACHINE_CODE_LENGTH_PER_INSTRUCTION * 2 >= self.result.text_section.len() {
```

Comment:
It is high yes, but it can only run into the end one sBPF instruction earlier. And that should be covered by `MAX_EMPTY_PROGRAM_MACHINE_CODE_LENGTH`. Generally I would be rather conservative and overestimate here as this is the only bounds-check for the emitted code size.